File: src/features/comments/deleteAccComment.ts
import { FastifyRequest, FastifyReply } from 'fastify'
import { verifyGJP } from '../../lib/tools'
import { query } from '../../lib/db'
import Logger from '../../lib/logger'

type Body = {
    accountID: number,
    gjp: string,
    commentID: string
}

export default async function handler(req: FastifyRequest<{ Body: Body }>, rep: FastifyReply) {
    if(!req.body.accountID || !req.body.gjp || !req.body.commentID) return -1

    if(!(await verifyGJP(req.body.accountID, req.body.gjp))) return -1

    await query("DELETE FROM acc_comments WHERE accCommentID = ? AND accountID = ? LIMIT 1", [req.body.commentID, req.body.accountID])
    
    Logger.event_delete('Account comment deleted')
    return 1
}

File: src/routes/comments/delete_acc_comment.rs
use actix_web::{Responder, HttpResponse, post, web::{Data, Form}};
use serde::Deserialize;
use sqlx::PgPool;

use crate::{models::account::Account, utils::encryption};

#[derive(Debug, Deserialize)]
struct Body {
    #[serde(rename = "accountID")]
    account_id: i64,
    gjp: String,
    #[serde(rename = "commentID")]
    comment_id: i64,
}

#[post("/deleteGJAccComment20.php")]
async fn handler(form: Form<Body>, db: Data<PgPool>) -> impl Responder {
    let account: Option<Account> = sqlx::query_as("SELECT * FROM accounts WHERE account_id = $1")
        .bind(&form.account_id)
        .fetch_optional(db.get_ref())
        .await.unwrap();

    if account.is_none() {
        return HttpResponse::Ok().body("-1");
    }

    let account_data: Account = account.unwrap();

    if !encryption::verify_gjp(account_data.password, form.gjp.clone()) {
        return HttpResponse::Ok().body("-1");
    }

    sqlx::query("DELETE FROM acc_comments WHERE account_comment_id = $1 AND account_id = $2")
        .bind(&form.comment_id)
        .bind(&account_data.account_id)
        .execute(db.get_ref())
        .await.unwrap();

    HttpResponse::Ok().body("1")
}